### DISCUSSION TITLE
Does IDOM Transpile/Transcompile to WebAsm/JS?

### DISCUSSION THREAD

**Discussion by Archmonger** [idom-team/idom#357] [Question] [answered]
Are `@idom.event`'s run within the Python backend, or is code prefixed with that decorator transcompiled to webasm/javascript and run on the client browser?

**Comment by rmorshea** [comment, marked as answer]
There is no transpiling, this was an explicit design choice I made in order to keep things simple. As a result almost everything happens server-side. I wrote [an article](https://ryanmorshead.com/articles/2021/idom-react-but-its-python/article/#virtual-document-object-model) about IDOM recently that might give you a better high level sense for how it works. If you're curious there's [VDOM spec](https://idom-docs.herokuapp.com/docs/specifications.html#vdom-mimetype) that describes exactly what is communicated to the client.
